### PR TITLE
Add Update endpoint to BooksController and clean up file

### PR DESCRIPTION
**Pull request by thiagograndesso**

### CODE REVIEW

File: src/VintageBookshelf.Api/Configuration/ApiConfig.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+using Microsoft.Extensions.DependencyInjection;
+
+namespace VintageBookshelf.Api.Configuration
+{
+    public static class ApiConfig
+    {
+        public static IServiceCollection AddWebApiConfig(this IServiceCollection services)
+        {
+            services.AddControllers();
+            services.Configure<ApiBehaviorOptions>(options => options.SuppressModelStateInvalidFilter = true);
```
Comment (dieguezthiago) [inline src/VintageBookshelf.Api/Configuration/ApiConfig.cs:12]: What do you have in mind on doing that?

File: src/VintageBookshelf.Api/Controllers/BooksController.cs [hunk 2]
```diff
@@ -49,17 +49,57 @@ public async Task<ActionResult<IEnumerable<BookDto>>> Add([FromBody] BookDto boo
             return CustomResponse(bookDto);
         }
         
+        [HttpPut("{id:long}")]
+        [ProducesResponseType(200)]
+        [ProducesResponseType(400)]
```
Comment (dieguezthiago) [inline src/VintageBookshelf.Api/Controllers/BooksController.cs:54]: What about 404 (NotFound)?

File: src/VintageBookshelf.Api/Controllers/BooksController.cs [hunk 3]
```diff
@@ -49,17 +49,57 @@ public async Task<ActionResult<IEnumerable<BookDto>>> Add([FromBody] BookDto boo
...
+        public async Task<ActionResult<BookDto>> Update(long id, [FromBody] BookDto bookDto)
+        {
```
Comment (dieguezthiago) [inline src/VintageBookshelf.Api/Controllers/BooksController.cs:57]: I believe you don't need to use [FromBody] in this case. The object would be used for that even without the attribute.

File: src/VintageBookshelf.Api/Controllers/BooksController.cs [hunk 4]
```diff
@@ -68,16 +108,18 @@ public async Task<ActionResult<IEnumerable<BookDto>>> GetById(long id)
...
             {
                 return NotFound();
             }
 
             await _bookService.Remove(id);
-            return CustomResponse(book);
+            return CustomResponse();
         }
 
         private async Task<bool> UploadImage(BookDto bookDto)
```
Comment (dieguezthiago) [inline src/VintageBookshelf.Api/Controllers/BooksController.cs:126]: For async methods (which are not controller actions) I would use the Async suffix at the end of the method name.

File: src/VintageBookshelf.Api/Controllers/BooksController.cs (hunk 3 above)
Comment (thiagograndesso) [reply src/VintageBookshelf.Api/Controllers/BooksController.cs:57]: This is mostly for displaying on Swagger that a body payload is required for this endpoint as an input. If you don't use `FromBody` attribute, then it won't be displayed on Swagger. Perhaps there's another option for this instead?

File: src/VintageBookshelf.Api/Controllers/BooksController.cs (hunk 4 above)
Comment (thiagograndesso) [reply src/VintageBookshelf.Api/Controllers/BooksController.cs:126]: Sure. I mean, it's mostly defined by usage convention. Do we want to rename all methods to follow this pattern? I believe that Microsoft added the `Async` suffix naming pattern for several library methods that were converted from the synchronous version after the introduction of `TPL` (Task Parallel Library) with the known `async` and `await` keywords. E.g: If there's a synchronous method named `UploadImage` in a .NET framework dll, then Microsoft named it `UploadImageAsync`.

File: src/VintageBookshelf.Api/Configuration/ApiConfig.cs (hunk 1 above)
Comment (thiagograndesso) [reply src/VintageBookshelf.Api/Configuration/ApiConfig.cs:12]: Model State is validated by the ASP.NET Core framework by default if not disabled. I want to have more control over the way that the Model State is validated. This means that I'll always receive incoming requests to controller's endpoints.

Comment (dieguezthiago) [reply src/VintageBookshelf.Api/Configuration/ApiConfig.cs:12]: So you want to check if the ModelState is valid on your code?

File: src/VintageBookshelf.Api/Controllers/BooksController.cs (hunk 3 above)
Comment (dieguezthiago) [reply src/VintageBookshelf.Api/Controllers/BooksController.cs:57]: Not sure. I think it is worth investigating once the framework has its pre conditions to associate the request with the parameters in the action.

File: src/VintageBookshelf.Api/Configuration/ApiConfig.cs (hunk 1 above)
Comment (thiagograndesso) [reply src/VintageBookshelf.Api/Configuration/ApiConfig.cs:12]: Yes, I want flexibility on my end to be able to return custom error messages. For example, if we let AP.NET Core to control Model State validation, default error messages will be returned instead.

File: src/VintageBookshelf.Api/Controllers/BooksController.cs (hunk 3 above)
Comment (thiagograndesso) [reply src/VintageBookshelf.Api/Controllers/BooksController.cs:57]: It's not an issue at the code level, I agree. It's mostly for documentation purposes and to show the correct body input content type (`application/json`).

File: src/VintageBookshelf.Api/Configuration/ApiConfig.cs (hunk 1 above)
Comment (dieguezthiago) [reply src/VintageBookshelf.Api/Configuration/ApiConfig.cs:12]: Yopu can check this one out then: 

```
services.AddControllers().
.ConfigureApiBehaviorOptions(
setupAction => { setupAction.InvalidModelStateResponseFactory = context => { **<CUSTOM RESPONSE FACTORY>** } } );
```

Comment (dieguezthiago) [reply src/VintageBookshelf.Api/Configuration/ApiConfig.cs:12]: I am not sure you can use it in MVC though.

**Pull request merged**